frontend2/src/api/helpers.ts

```diff
@@ -84,3 +86,42 @@ export const prefetchNextPage = async <
     }
   }
 };
+
+const safeLoaderHelper = async <T, K>(
+  request: T,
+  factory: QueryFactory<T, K> | PaginatedQueryFactory<T, K>,
+  queryClient: QueryClient,
+): Promise<void> => {
+  try {
+    await queryClient.ensureQueryData({
+      queryKey: buildKey(factory.queryKey, request),
+      queryFn: async () => await factory.queryFn(request, queryClient, true),
+    });
+  } catch (err) {
+    if ((err as ResponseError).response.status > 500) {
```

Comment:
`(err as ResponseError).response.status` errors if `err.response` is undefined, which can happen when err is not a ResponseError